frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx

```diff
@@ -188,22 +245,28 @@ export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> =
       },
     };
 
-    return (props.operatorGroup.data.some(
-      (group) => group.metadata.namespace === selectedTargetNamespace,
-    )
-      ? Promise.resolve()
-      : k8sCreate(OperatorGroupModel, operatorGroup)
-    )
-      .then(() => k8sCreate(SubscriptionModel, subscription))
-      .then(() =>
-        history.push(
-          resourceListPathFromModel(
-            ClusterServiceVersionModel,
-            targetNamespace || props.targetNamespace || selectedTargetNamespace,
-          ),
-        ),
-      )
-      .catch(({ message = 'Could not create operator subscription.' }) => setError(message));
+    return createNamespaceIfNecessary().then(
+      () => {
+        (props.operatorGroup.data.some(
+          (group) => group.metadata.namespace === selectedTargetNamespace,
+        )
+          ? Promise.resolve()
+          : k8sCreate(OperatorGroupModel, operatorGroup)
+        )
+          .then(() => k8sCreate(SubscriptionModel, subscription))
+          .then(() =>
+            history.push(
+              resourceListPathFromModel(
+                ClusterServiceVersionModel,
+                targetNamespace || props.targetNamespace || selectedTargetNamespace,
+              ),
+            ),
+          )
+          .catch(({ message = 'Could not create operator subscription.' }) => setError(message));
+
+      },
+      (err) => setError(err.message),
```

Comment:
Better to use `catch`

```suggestion
      .catch(({ message = 'Could not create namespace.' }) => setError(message),
```